### PR TITLE
Add type annotations so that static code analysis and code completion work properly

### PR DESCRIPTION
**Pull request by sjinks**
## Description

This PR adds type annotations to the files I have worked recently with so that static code analysis and code completion features work properly.

### Unit Tests

Add type annotations

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

N/A

### CODE REVIEW

File: search/includes/classes/class-health.php [hunk 1]
```diff
@@ -131,7 +135,7 @@ public function validate_index_entity_count( array $query_args, \ElasticPress\In
 	 *      'post_status' => array( $post_statuses )
 	 * ];
 	 *
-	 * @param mixed $indexable Instance of an ElasticPress Indexable Object to search on
+	 * @param Comment|Post|Term|User $indexable Instance of an ElasticPress Indexable Object to search on
```
Comment (pschoffer) [inline search/includes/classes/class-health.php:138]: Should this be `Indexable` instead?

**Review by pschoffer — CHANGES_REQUESTED** [review]

File: search/includes/classes/class-health.php (hunk 1 above)
Comment (sjinks) [reply search/includes/classes/class-health.php:138]: I am afraid it shouldn't — `Indexable` does not have a `format_args()` method used by the code. That's why I had to enumerate all classes that have.

Comment (sjinks) [reply search/includes/classes/class-health.php:138]: https://github.com/Automattic/vip-go-mu-plugins/blob/2a822a6f5202daaf9085d7d45acdac0d69ae2fd3/search/includes/classes/class-health.php#L145

Comment (pschoffer) [reply search/includes/classes/class-health.php:138]: Hmm I see. Probably it should though? Maybe it should be an abstract class defining `format_args` as an abstract method. That it is how it is being used anyways I feel like, what do you think?

Comment (sjinks) [reply search/includes/classes/class-health.php:138]: I will address this in another PR: I saw several places where `Indexable` does not fit while working with the Search plugin; chances are that `format_args()` is not the only method.

Comment (sjinks) [reply search/includes/classes/class-health.php:138]: Actually, I am not sure this is even possible — those classes are defined in a third-party module, we cannot even attach an interface to them :-( The upstream will need to fix this.

**Review by pschoffer — APPROVED** [review]
LGTM

**Comment by sjinks** [comment]
r1998-stacks

**Pull request merged**